pkg/network/netlink/conntracker.go

```diff
@@ -232,14 +232,16 @@ func (ctr *realConntracker) register(c Con) int {
 	}
 
 	then := time.Now().UnixNano()
-	defer func() {
-		atomic.AddInt64(&ctr.stats.registers, 2)
-		atomic.AddInt64(&ctr.stats.registersTotalTime, time.Now().UnixNano()-then)
-	}()
 
 	ctr.Lock()
 	defer ctr.Unlock()
 
+	l := ctr.cache.Len()
+	defer func() {
+		atomic.AddInt64(&ctr.stats.registers, int64(ctr.cache.Len()-l))
+		atomic.AddInt64(&ctr.stats.registersTotalTime, time.Now().UnixNano()-then)
+	}()
+
 	ctr.cache.Add(c, true)
```

Comment:
Does this need to be done in a defer at all? This function has almost no branching, and it doesn't really improve readability to handle stats in a defer.